File: mybatis-plus-join-test/test-join/src/main/java/com/github/yulichang/test/join/entity/AddressGeneric.java
package com.github.yulichang.test.join.entity;


import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import lombok.Data;

@Data
public class AddressGeneric<ID, USER_ID, AREA_ID, TEL, ADDRESS, DEL> {

    @TableId
    private ID id;

    private USER_ID userId;

    private AREA_ID areaId;

    private TEL tel;

    private ADDRESS address;

    @TableLogic
    private DEL del;
}

File: mybatis-plus-join-test/test-kotlin/src/main/java/com/github/yulichang/test/kt/entity/AddressGeneric.kt
package com.github.yulichang.test.kt.entity

import com.baomidou.mybatisplus.annotation.TableId
import com.baomidou.mybatisplus.annotation.TableLogic

open class AddressGeneric<ID, USER_ID, AREA_ID, TEL, ADDRESS, DEL> {
    @TableId
    var id: ID? = null
    var userId: USER_ID? = null
    var areaId: AREA_ID? = null
    var tel: TEL? = null
    var address: ADDRESS? = null

    @TableLogic
    var del: DEL? = null
}
